pkgs/servers/sql/postgresql/default.nix

```diff
@@ -102,7 +113,12 @@ let
       (if atLeast "16" then ./patches/disable-normalize_exec_path.patch
        else ./patches/disable-resolve_symlinks.patch)
       ./patches/less-is-more.patch
-      ./patches/hardcode-pgxs-path.patch
+
+      # Hardcode the path to pgxs and other dirs so that pg_config returns the path in the package,
+      # rather than one relative to the location pg_config was executed in.
+      # The placeholders are resolved in postPatch.
+      ./patches/hardcode-pg_config-paths.patch
```

Comment:
Since a relative path is used for most of the values here, isn't this orthogonal to the rest of the changes? Asking because if all of that needs more discussion, this could be introduced first on its own.

Reply:
I guess it could be split to a separate commit (using `@out@` instead of `@dev@` and then this commit would just replace the output placeholder) but I do not see much of a point. This offers no benefit without the output splitting.

Reply:
Oh you're right, since the outputs are split, it's not guaranteed that each path is relative to `$out` :+1: 

Sorry, I agree with the way this was done in that case.